src/main/java/cz/student/shaccson/UserDao.java

```diff
@@ -0,0 +1,104 @@
+package cz.student.shaccson;
+
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDao {
+    private final Connection conn;
+
+    public UserDao(final Connection conn) {
+        this.conn = conn;
+    }
+
+    public void addUser(final User user) {
+        final String query = "INSERT INTO users (email, username, password) VALUES (?, ?, ?)";
+        try (final PreparedStatement statement = conn.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getEmail());
+            statement.setString(2, user.getUsername());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.executeUpdate();
+
+            final ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                user.setId(generatedKeys.getInt(1));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```

Comment:
CLEAN-CODE: zlepsit logovani chybovych stavu na vsech mistech.

```suggestion
            System.err.println("Problem with creating user in database: " + e.getMessage());
            e.printStackTrace(System.err);
            return null;
```

Reply:
pridano lepsi logovani chybovych stavu